packages/backend/src/handlers/actions/quests/createCompletion/createCompletion.ts

```diff
@@ -0,0 +1,67 @@
+import { CreateQuestCompletionInput, CreateQuestCompletionOutput,
+  Quest_Completion_Insert_Input,
+  QuestRepetition_Enum,
+  QuestStatus_Enum, } from '../../../../lib/autogen/hasura-sdk';
+import { client } from '../../../../lib/hasuraClient';
+
+export async function createCompletion(
+  playerId: string,
+  questCompletion: CreateQuestCompletionInput,
+): Promise<CreateQuestCompletionOutput> {
+  if (!questCompletion.submission_link && !questCompletion.submission_text) {
+    throw new Error('Must provide at least a submission link or text');
+  }
+
+  const { quest_by_pk: quest } = await client.GetQuestById({ quest_id: questCompletion.quest_id });
+  if (!quest) {
+    throw new Error('Quest not found');
+  }
+
+  if (quest.status !== QuestStatus_Enum.Open) {
+    throw new Error('Quest must be open');
+  }
+
+  // Personal or unique, check if not already done by player
+  if (quest.repetition === QuestRepetition_Enum.Unique || quest.repetition === QuestRepetition_Enum.Personal) {
```

Comment:
If its a unique quest, should we check if it was already completed by _any_ player instead of only checking if it was completed by the player calling the mutation?

Reply:
Here, the quest is still OPEN, so:
If it's UNIQUE, no completion has been accepted yet, so anyone can submit ONLY one
If it's PERSONAL, player can only submit one.